src/Exception/CompositeException.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace React\Promise\Exception;
+
+class CompositeException extends \Exception
+{
+    private $exceptions;
+
+    public function __construct(array $exceptions, $message = '', $code = 0, $previous = null)
+    {
+        parent::__construct($message, $code, $previous);
+
+        $this->exceptions = $exceptions;
+    }
+
+    /**
+     * @return \Throwable[]|\Exception[]
+     */
+    public function getExceptions()
+    {
+        return $this->exceptions;
+    }
+
+    public static function tooManyPromisesRejected(array $reasons)
```

Comment:
`@internal`?

Reply:
I think something like that makes also sense for custom combinator functions.

Reply:
Name constructor removed in df0976e.